pkg/download/dependency_resolution/resolver/basic_dep_resolver.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resolver
+
+import (
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/maps"
+	config "github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/coordinate"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter/reference"
+	"strings"
+)
+
+type basicResolver struct {
+	configsById map[string]config.Config
+}
+
+func BasicResolver(configsById map[string]config.Config) basicResolver {
+	return basicResolver{
+		configsById: configsById,
+	}
+}
+
+func (r basicResolver) ResolveDependencyReferences(configToBeUpdated *config.Config) {
+	resolveScope(configToBeUpdated, r.configsById)
+	resolveTemplateB(configToBeUpdated, r.configsById)
```

Comment:
```suggestion
	resolveTemplate(configToBeUpdated, r.configsById)
```

Reply:
I've lazily put both resolvers in the same package for now, so while they share the `resolveScope` they each implement a`resolveTemplate` which would overlap.

I'm not really happy with the package structure, but I didn't want to introduce several packages with a single file each

Reply:
ah ok looked like a typo (left-over) from something to experiment with... anyway , ok i guess :man_shrugging:

Reply:
I've renamed it to something more intentionally different :)